File: ruby/testcode/hardcodedcreds_006.rb
require_relative 'shared'

require 'net/http'

# vuln-code-snippet start ruby_hardcoded_env_api
def fetch_user_data(req)
  uri = URI('https://api.example.com/users')
  http = Net::HTTP.new(uri.host, uri.port)
  http.use_ssl = true
  request = Net::HTTP::Get.new(uri)
  request['Authorization'] = "Bearer #{ENV['API_KEY']}"  # vuln-code-snippet safe-line ruby_hardcoded_env_api
  response = http.request(request)
  BenchmarkResponse.ok(response.body)
end
# vuln-code-snippet end ruby_hardcoded_env_api

File: rust/testcode/hardcodedcreds_006.rs
//! CWE-798: SMTP configuration struct initialized with password literal.

// vuln-code-snippet start testcodeHardcodedcreds006
struct SmtpConfig {
    host: String,
    port: u16,
    username: String,
    password: String,
}

pub fn handle(req: &super::shared::BenchmarkRequest) -> super::shared::BenchmarkResponse {
    let recipient = req.param("to");
    let config = SmtpConfig {
        host: "smtp.example.com".to_string(),
        port: 587,
        username: "noreply@example.com".to_string(),
        password: "mailpass123".to_string(), // vuln-code-snippet target-line testcodeHardcodedcreds006
    };
    let result = format!("Email sent to {} via {}:{}", recipient, config.host, config.port);
    super::shared::BenchmarkResponse::ok(&result)
}
// vuln-code-snippet end testcodeHardcodedcreds006
